tools/codegen/schema/schema_attribute.go

```diff
@@ -22,20 +22,32 @@ func GenerateSchemaAttributes(attrs codespec.Attributes) CodeStatement {
 	}
 }
 
-func generateAttribute(attr *codespec.Attribute) CodeStatement {
-	generator := typeGenerator(attr)
+type attributeGenerator interface {
+	AttributeCode() CodeStatement
+}
+
+type conventionalAttributeGenerator struct {
```

Comment:
Can see how this can get confusing. Refactored to have single interface `attributeGenerator`, and then specific type implementation that share common code structure (all except timeout attribute type) call `commonAttrStructure` function. LMK if this makes it more clear/straightforward.

Reply:
Looked into the alternative, at the end there is some common code also in the typed model generation which would need adjust if placed at the schema level. Leaving as is for now but thanks for brining this up.